docs/src/main/asciidoc/security-authorize-web-endpoints-reference.adoc

```diff
@@ -386,6 +386,186 @@ It is possible to use multiple expressions in the role definition.
 <3> This `/subject/user` endpoint requires an authenticated user that has been granted the role "User" through the use of the `@RolesAllowed("${customer:User}")` annotation, as we did not set the configuration property `customer`.
 <4> This `/subject/secured` endpoint requires an authenticated user that has been granted the role `User` in production but allows any authenticated user in development mode.
 
+=== Permission annotation
+
+Quarkus also provides the `io.quarkus.security.PermissionsAllowed` annotation that will permit any authenticated user with given permission to access the resource.
+The annotation is extension of the common security annotations and there is no relation to configuration permissions defined with the configuration property `quarkus.http.auth.permission`.
+
+.Example of endpoints secured with a `@PermissionsAllowed` annotation
+
+[source,java]
+----
+import io.quarkus.arc.Arc;
+import io.vertx.ext.web.RoutingContext;
+import jakarta.ws.rs.GET;
+import jakarta.ws.rs.POST;
+import jakarta.ws.rs.Path;
+import jakarta.ws.rs.QueryParam;
+
+import io.quarkus.security.PermissionsAllowed;
+
+import java.security.Permission;
+import java.util.Collection;
+import java.util.Collections;
+
+@Path("/crud")
+public class CRUDResource {
+
+    @PermissionsAllowed("create") <1>
+    @PermissionsAllowed("update")
```

Comment:
What is your opinion on `@PermissionAllowed` having an attribute, which by default, will require that at least one of the permissions is allowed, but can be set to enforce that all of them are allowed ?
Supporting more than `@PermissionAllowed` annotation can still be allowed to support some complex variations like:
```
// Either a or b or c is allowed and either d or e or f is allowed
@PermissionsAllowed("a,b,c")
@PermissionsAllowed("d,e,f")
``` 
but for a fairly typical case where 2 permissions must be allowed, one would say something like:
```
@PermissionsAllowed(value= {"d","e","f"}, inclusive=true)
```
or something like that, which can make some difference re the annotation noise if a lot of methods are annotated. We can do it in a follow up PR though if you agree

Reply:
Done, great idea!